src/execution/compiler/operator/static_aggregation_translator.cpp

```diff
@@ -22,6 +22,8 @@ StaticAggregationTranslator::StaticAggregationTranslator(const planner::Aggregat
       build_pipeline_(this, Pipeline::Parallelism::Parallel) {
   TERRIER_ASSERT(plan.GetGroupByTerms().empty(), "Global aggregations shouldn't have grouping keys");
   TERRIER_ASSERT(plan.GetChildrenSize() == 1, "Global aggregations should only have one child");
+  build_pipeline_.UpdateParallelism(Pipeline::Parallelism::Serial);
```

Comment:
@lmwnshn Thoughts

Without this, the following query sequence:
```
create table t (a INT, b INT, c INT);
select sum(a), sum(b), sum(c) from t
```

produces the error:
```
[2020-08-19 06:25:02.364] [execution_logger] [error] ERROR IN MODULE:
Stored value type does not match pointer operand type!
  store %1* %26, %"class.terrier::execution::sql::IntegerSumAggregate"** %5
 %"class.terrier::execution::sql::IntegerSumAggregate"*Stored value type does not match pointer operand type!
  store %1* %42, %"class.terrier::execution::sql::IntegerSumAggregate"** %8
 %"class.terrier::execution::sql::IntegerSumAggregate"*
```

Reply:
```
cmudb@cmudb-NUC10i7FNH ~/CLionProjects/terrier
 % psql -h localhost -p 15721 -d terrier
psql (12.2 (Ubuntu 12.2-4), server 9.5devel)
Type "help" for help.

terrier=# create table t (a INT, b INT, c INT);
CREATE TABLE
terrier=# select sum(a), sum(b), sum(c) from t;
 ?column? | ?column? | ?column? 
----------+----------+----------
          |          |         
(1 row)

terrier=# insert into t values (1,2,3);
INSERT 0 1
terrier=# select sum(a), sum(b), sum(c) from t;
 ?column? | ?column? | ?column? 
----------+----------+----------
        1 |        2 |        3
(1 row)

terrier=# \q
```

Unable to reproduce on master, trying your branch now.

Reply:
Oh um...i forgot to mention, you need to run the queries in compiled mode (not interpret).

Reply:
Not seeing it on your branch either actually..

Reply:
Oh ok

Reply:
Reproduced, the buggy TPL code:
```
struct OutputStruct {
    out0: Integer
    out1: Integer
    out2: Integer
}
struct AggPayload {
    agg_term_attr0: IntegerSumAggregate
    agg_term_attr1: IntegerSumAggregate
    agg_term_attr2: IntegerSumAggregate
}
struct AggValues {
    agg_term_attr0: Integer
    agg_term_attr1: Integer
    agg_term_attr2: Integer
}
struct QueryState {
    execCtx: *ExecutionContext
    aggs   : AggPayload
}
struct P2_State {
    aggs1: AggPayload
}
struct P1_State {
    
}
fun MergeAggregates(queryState: *QueryState, pipelineState: *P2_State) -> nil {
    @aggMerge(&queryState.aggs.agg_term_attr0, &pipelineState.aggs1.agg_term_attr0)
    @aggMerge(&queryState.aggs.agg_term_attr1, &pipelineState.aggs1.agg_term_attr1)
    @aggMerge(&queryState.aggs.agg_term_attr2, &pipelineState.aggs1.agg_term_attr2)
    return
}

fun Query1_Init(queryState: *QueryState) -> nil {
    return
}

fun Query1_Pipeline2_InitPipelineState(queryState: *QueryState, pipelineState: *P2_State) -> nil {
    @aggInit(&pipelineState.aggs1.agg_term_attr0)
    @aggInit(&pipelineState.aggs1.agg_term_attr1)
    @aggInit(&pipelineState.aggs1.agg_term_attr2)
    return
}

fun Query1_Pipeline2_TearDownPipelineState(queryState: *QueryState, pipelineState: *P2_State) -> nil {
    return
}

fun Query1_Pipeline2_ParallelWork(queryState: *QueryState, pipelineState: *P2_State, tvi: *TableVectorIterator) -> nil {
    var slot: TupleSlot
    for (@tableIterAdvance(tvi)) {
        var vpi = @tableIterGetVPI(tvi)
        for (; @vpiHasNext(vpi); @vpiAdvance(vpi)) {
            slot = @vpiGetSlot(vpi)
            var aggValues: AggValues
            aggValues.agg_term_attr0 = @vpiGetIntNull(vpi, 0)
            aggValues.agg_term_attr1 = @vpiGetIntNull(vpi, 1)
            aggValues.agg_term_attr2 = @vpiGetIntNull(vpi, 2)
            @aggAdvance(&pipelineState.aggs1.agg_term_attr0, &aggValues.agg_term_attr0)
            @aggAdvance(&pipelineState.aggs1.agg_term_attr1, &aggValues.agg_term_attr1)
            @aggAdvance(&pipelineState.aggs1.agg_term_attr2, &aggValues.agg_term_attr2)
        }
    }
    return
}

fun Query1_Pipeline2_Init(queryState: *QueryState) -> nil {
    var threadStateContainer = @execCtxGetTLS(queryState.execCtx)
    @tlsReset(threadStateContainer, @sizeOf(P2_State), Query1_Pipeline2_InitPipelineState, Query1_Pipeline2_TearDownPipelineState, queryState)
    return
}

fun Query1_Pipeline2_Run(queryState: *QueryState) -> nil {
    @aggInit(&queryState.aggs.agg_term_attr0)
    @aggInit(&queryState.aggs.agg_term_attr1)
    @aggInit(&queryState.aggs.agg_term_attr2)
    var col_oids: [3]uint32
    col_oids[0] = 1
    col_oids[1] = 2
    col_oids[2] = 3
    @iterateTableParallel(1002, col_oids, queryState, queryState.execCtx, Query1_Pipeline2_ParallelWork)
    @tlsIterate(@execCtxGetTLS(queryState.execCtx), queryState, MergeAggregates)
    return
}

fun Query1_Pipeline2_TearDown(queryState: *QueryState) -> nil {
    @tlsClear(@execCtxGetTLS(queryState.execCtx))
    return
}

fun Query1_Pipeline1_InitPipelineState(queryState: *QueryState, pipelineState: *P1_State) -> nil {
    return
}

fun Query1_Pipeline1_TearDownPipelineState(queryState: *QueryState, pipelineState: *P1_State) -> nil {
    return
}

fun Query1_Pipeline1_SerialWork(queryState: *QueryState, pipelineState: *P1_State) -> nil {
    var aggRow = &queryState.aggs
    var outRow = @ptrCast(*OutputStruct, @resultBufferAllocRow(queryState.execCtx))
    outRow.out0 = @aggResult(&aggRow.agg_term_attr0)
    outRow.out1 = @aggResult(&aggRow.agg_term_attr1)
    outRow.out2 = @aggResult(&aggRow.agg_term_attr2)
    return
}

fun Query1_Pipeline1_Init(queryState: *QueryState) -> nil {
    var threadStateContainer = @execCtxGetTLS(queryState.execCtx)
    @tlsReset(threadStateContainer, @sizeOf(P1_State), Query1_Pipeline1_InitPipelineState, Query1_Pipeline1_TearDownPipelineState, queryState)
    return
}

fun Query1_Pipeline1_Run(queryState: *QueryState) -> nil {
    var pipelineState = @ptrCast(*P1_State, @tlsGetCurrentThreadState(@execCtxGetTLS(queryState.execCtx)))
    @execCtxStartResourceTracker(queryState.execCtx, 4)
    Query1_Pipeline1_SerialWork(queryState, pipelineState)
    @resultBufferFinalize(queryState.execCtx)
    @execCtxEndPipelineTracker(queryState.execCtx, 1, 1)
    return
}

fun Query1_Pipeline1_TearDown(queryState: *QueryState) -> nil {
    @tlsClear(@execCtxGetTLS(queryState.execCtx))
    return
}

fun Query1_TearDown(queryState: *QueryState) -> nil {
    return
}


[2020-08-19 11:08:18.664] [execution_logger] [error] ERROR IN MODULE:
Stored value type does not match pointer operand type!
  store %1* %16, %"class.terrier::execution::sql::IntegerSumAggregate"** %5
 %"class.terrier::execution::sql::IntegerSumAggregate"*Stored value type does not match pointer operand type!
  store %1* %22, %"class.terrier::execution::sql::IntegerSumAggregate"** %8
 %"class.terrier::execution::sql::IntegerSumAggregate"*Stored value type does not match pointer operand type!
  store %1* %5, %"class.terrier::execution::sql::IntegerSumAggregate"** %2
 %"class.terrier::execution::sql::IntegerSumAggregate"*Stored value type does not match pointer operand type!
  store %1* %7, %"class.terrier::execution::sql::IntegerSumAggregate"** %3
 %"class.terrier::execution::sql::IntegerSumAggregate"*Stored value type does not match pointer operand type!
  store %1* %30, %"class.terrier::execution::sql::IntegerSumAggregate"** %10
 %"class.terrier::execution::sql::IntegerSumAggregate"*Stored value type does not match pointer operand type!
  store %1* %34, %"class.terrier::execution::sql::IntegerSumAggregate"** %12
 %"class.terrier::execution::sql::IntegerSumAggregate"*
```

By forcing it to be serial:
```
struct AggPayload {
    agg_term_attr0: IntegerSumAggregate
    agg_term_attr1: IntegerSumAggregate
    agg_term_attr2: IntegerSumAggregate
}
struct AggValues {
    agg_term_attr0: Integer
    agg_term_attr1: Integer
    agg_term_attr2: Integer
}
struct OutputStruct {
    out0: Integer
    out1: Integer
    out2: Integer
}
struct QueryState {
    execCtx: *ExecutionContext
    aggs   : AggPayload
}
struct P2_State {
    
}
struct P1_State {
    
}
fun Query2_Init(queryState: *QueryState) -> nil {
    return
}

fun Query2_Pipeline2_InitPipelineState(queryState: *QueryState, pipelineState: *P2_State) -> nil {
    return
}

fun Query2_Pipeline2_TearDownPipelineState(queryState: *QueryState, pipelineState: *P2_State) -> nil {
    return
}

fun Query2_Pipeline2_SerialWork(queryState: *QueryState, pipelineState: *P2_State) -> nil {
    var tviBase: TableVectorIterator
    var tvi = &tviBase
    var col_oids: [3]uint32
    col_oids[0] = 1
    col_oids[1] = 2
    col_oids[2] = 3
    @tableIterInit(tvi, queryState.execCtx, 1002, col_oids)
    var slot: TupleSlot
    for (@tableIterAdvance(tvi)) {
        var vpi = @tableIterGetVPI(tvi)
        for (; @vpiHasNext(vpi); @vpiAdvance(vpi)) {
            slot = @vpiGetSlot(vpi)
            var aggValues: AggValues
            aggValues.agg_term_attr0 = @vpiGetIntNull(vpi, 0)
            aggValues.agg_term_attr1 = @vpiGetIntNull(vpi, 1)
            aggValues.agg_term_attr2 = @vpiGetIntNull(vpi, 2)
            @aggAdvance(&queryState.aggs.agg_term_attr0, &aggValues.agg_term_attr0)
            @aggAdvance(&queryState.aggs.agg_term_attr1, &aggValues.agg_term_attr1)
            @aggAdvance(&queryState.aggs.agg_term_attr2, &aggValues.agg_term_attr2)
        }
    }
    @tableIterClose(tvi)
    return
}

fun Query2_Pipeline2_Init(queryState: *QueryState) -> nil {
    var threadStateContainer = @execCtxGetTLS(queryState.execCtx)
    @tlsReset(threadStateContainer, @sizeOf(P2_State), Query2_Pipeline2_InitPipelineState, Query2_Pipeline2_TearDownPipelineState, queryState)
    return
}

fun Query2_Pipeline2_Run(queryState: *QueryState) -> nil {
    @aggInit(&queryState.aggs.agg_term_attr0)
    @aggInit(&queryState.aggs.agg_term_attr1)
    @aggInit(&queryState.aggs.agg_term_attr2)
    var pipelineState = @ptrCast(*P2_State, @tlsGetCurrentThreadState(@execCtxGetTLS(queryState.execCtx)))
    @execCtxStartResourceTracker(queryState.execCtx, 4)
    Query2_Pipeline2_SerialWork(queryState, pipelineState)
    @execCtxEndPipelineTracker(queryState.execCtx, 2, 2)
    return
}

fun Query2_Pipeline2_TearDown(queryState: *QueryState) -> nil {
    @tlsClear(@execCtxGetTLS(queryState.execCtx))
    return
}

fun Query2_Pipeline1_InitPipelineState(queryState: *QueryState, pipelineState: *P1_State) -> nil {
    return
}

fun Query2_Pipeline1_TearDownPipelineState(queryState: *QueryState, pipelineState: *P1_State) -> nil {
    return
}

fun Query2_Pipeline1_SerialWork(queryState: *QueryState, pipelineState: *P1_State) -> nil {
    var aggRow = &queryState.aggs
    var outRow = @ptrCast(*OutputStruct, @resultBufferAllocRow(queryState.execCtx))
    outRow.out0 = @aggResult(&aggRow.agg_term_attr0)
    outRow.out1 = @aggResult(&aggRow.agg_term_attr1)
    outRow.out2 = @aggResult(&aggRow.agg_term_attr2)
    return
}

fun Query2_Pipeline1_Init(queryState: *QueryState) -> nil {
    var threadStateContainer = @execCtxGetTLS(queryState.execCtx)
    @tlsReset(threadStateContainer, @sizeOf(P1_State), Query2_Pipeline1_InitPipelineState, Query2_Pipeline1_TearDownPipelineState, queryState)
    return
}

fun Query2_Pipeline1_Run(queryState: *QueryState) -> nil {
    var pipelineState = @ptrCast(*P1_State, @tlsGetCurrentThreadState(@execCtxGetTLS(queryState.execCtx)))
    @execCtxStartResourceTracker(queryState.execCtx, 4)
    Query2_Pipeline1_SerialWork(queryState, pipelineState)
    @resultBufferFinalize(queryState.execCtx)
    @execCtxEndPipelineTracker(queryState.execCtx, 2, 1)
    return
}

fun Query2_Pipeline1_TearDown(queryState: *QueryState) -> nil {
    @tlsClear(@execCtxGetTLS(queryState.execCtx))
    return
}

fun Query2_TearDown(queryState: *QueryState) -> nil {
    return
}
```

LGTM for now. Will open an issue.